app/presenters/history_event.rb

```diff
@@ -0,0 +1,47 @@
+require "delegate"
+
+# HistoryEvent wraps & delegates to C2Version
+class HistoryEvent < SimpleDelegator
+  include ActiveModel::Conversion
+
+  def event_type
+    model.event
+  end
+
+  def comment_text
+    # For some reason, paper_trail's "reify" doesn't work with comment versions
+    Comment.find(item_id).comment_text
+  end
+
+  def attachment
+    # "reify" not working with attachment versions either
+    Attachment.find(item_id)
+  end
+
+  def decorate_item
+    reify.decorate
+  end
+
+  def model
+    __getobj__
+  end
+
+  def to_partial_path
+    "proposals/details/history/" + partial_name
+  end
+
+  private
+
+  def partial_name
+    case item_type
```

Comment:
instead of this case statement we can do:

```ruby
item_type.split("::").last
```

...I think this will work for all cases :)